test/integration/variable_test.rb

```diff
@@ -79,4 +79,10 @@ def test_hash_with_default_proc
   def test_multiline_variable
     assert_equal 'worked', Template.parse("{{\ntest\n}}").render!('test' => 'worked')
   end
+
+  def test_global_options_on_variable
+    template = Template.parse(%({{test}}), global_filters: ['truncate: 5'])
```

Comment:
It isn't needed during parsing, just during rendering, so it would make sense to store the proc as part of the Liquid::Context.  So instead of [`taint_check(obj)` in Variable#render](https://github.com/Shopify/liquid/blob/7b4398d0c4f5b5b766677c8ad1ed159fed4cb041/lib/liquid/variable.rb#L77) you could do something like `context.global_variable_filter.call(obj)`.